### PR TITLE
change autocomplete for ajax requests

### PR DESCRIPTION
**Pull request by dellert**
Changed autocomplete plugin for use it with ajax. Docs updated too

### CODE REVIEW

**Comment by geminitydesigns** [comment]
Does this fix the ajax issue with the Materialize Autocomplete?

**Comment by dellert** [comment]
@geminitydesigns for me works good. I wrote in documentation how to use it

**Comment by geminitydesigns** [comment]
So can I pass in a json file for this?

**Comment by gbsuar** [comment]
Hi! In this version (#3694) we can use arrow keys in autocomplete. Is it intended to implement?
Thanks!!

**Comment by ReckerSwartz** [comment]
when you going to release it in offical update?

**Comment by aashay0101** [comment]
@dellert Your modification to the existing Autocomplete Plugin is really amazing, but wouldn't it be better if instead of changing the existing plugin, you make another **autocompleteajax** function that can be used for ajax purposes and the normal **ajaxcomplete** function can be used for autocomplete purposes only. This way the existing code written using the autocomplete function wont have to be altered and would also ensure the backwards compatibility for versions 0.97.7 and below.

So basically adding your plugin just below the existing plugin (**autocomplete**) with different name like **autocompleteajax**

You also wont have to remove the existing Markup in forms.html file, but only add the documentation for the new plugin. You get the gist.

**Comment by ahmetalpbalkan** [comment]
Does anybody have any interest in moving this forward? It looks like the community is in need of this.

**Comment by dellert** [comment]
So, get it here https://github.com/dellert/autocompleteajax

**Comment by kmmbvnr** [comment]
Btw, it's pretty easy to make [devbridge jquery-autocomplete](https://github.com/devbridge/jQuery-Autocomplete) work with MaterializeCSS, just with a few styles

```
.autocomplete-suggestions {
    background-color: #fff;
    box-shadow: 0 2px 2px 0 rgba(0,0,0,0.14), 0 1px 5px 0 rgba(0,0,0,0.12), 0 3px 1px -2px rgba(0,0,0,0.2);
    max-height: 650px;
    min-width: 100px;
    overflow-y: auto;
    will-change: width, height;
    z-index: 999;

    .autocomplete-suggestion,
    .autocomplete-no-suggestion {
        clear: both;
        color: #26a69a;
        color: rgba(0,0,0,0.87);
        cursor: pointer;
        display: block;
        font-size: 16px;
        line-height: 1.5rem;
        line-height: 22px;
        min-height: 50px;
        padding: 14px 16px;
        text-align: left;
        text-transform: none;
        width: 100%;

        &.autocomplete-selected {
            background-color: #eee;
        }
    }
}
```

**Comment by kmmbvnr** [comment]
And here is the Sample for the Chips integration

```html
<div id="autocomplete-chips">
<input type="hidden">
<select class="browser-default" style="dispaly:none">
  <option value="1" selected>Preselected 1</option>
  <option value="2" selected>Preselected 2</option>
</select>
<div class="chips"></div>
</div>
```

```javascript
  var initial_data = $('#autocomplete-chips select option').map(function() {
    return { tag:$(this).text(),  id:$(this).val() }
  })
  $('#autocomplete-chips .chips').material_chip({
    data: Array.from(initial_data)
  })
  $chips
    .on('chip.delete', function(e, chip){
       $(this).parent().find('select option')
         .filter(function() {return $(this).text() == chip.tag})
         .remove();
    })
    .on('chip.add', function(e, chip){
      var $selectData = $(this).parent().find('input[type="hidden"]'),
            data_id = $selectData.attr('data-id'),
            data_value = $selectData.attr('data-value')

      if(!data_id) {
              $(this).find('div')
                .filter(function(){return this.childNodes[0].nodeValue === chip.tag})
                .find('i').click()
            } else {
              $(this).parent().find('select').append(
                $("<option></option>")
                  .attr('selected','selected').attr("value", data_id).text(data_value))
              $selectData.attr('data-id', '').attr('data-value', '')
      }
    })
  $('#autocomplete-chips .chips input[class="input"]')
          .devbridgeAutocomplete({
            dataType: 'json',
            onSelect: function(suggestion) {
              var that = $(this);
              that.parent().parent().find('input[type="hidden"]')
                .attr("data-id", suggestion.data.id)
                .attr("data-value", suggestion.value);
              setTimeout(function() {
                e = jQuery.Event("keydown");
                e.which = 13;
                e.keyCode = 13;
                that.trigger(e);
              })
            },
            onInvalidateSelection: function() {
              $(this).parent().parent().find('input[type="hidden"]')
                .attr("data-id", '').attr("data-value", '');
            },
            serviceUrl: '...an url here...',
  })
```

**Comment by acburst** [comment]
Added updateData method for autocomplete v1.0.0 in bf24cb7d

**Comment by ghopretz** [comment]
> Btw, it's pretty easy to make [devbridge jquery-autocomplete](https://github.com/devbridge/jQuery-Autocomplete) work with MaterializeCSS
> 
> ```
> <div class="input-field col s12">
>   <input type="hidden" id="autocomplete-value" class="autocomplete">
>   <input type="text" id="autocomplete-input" class="autocomplete">
>   <label for="autocomplete-input">Autocomplete</label>
> </div>
> ```
> 
> ```js
> $('#autocomplete-input').devbridgeAutocomplete({
>     onSelect: function(suggestion) {
>         $('#autocomplete-value').val(suggestion.data.id)
>      },
>      onInvalidateSelection: function() {
>          $('#autocomplete-value').val('')
>       },
>       serviceUrl: '...an url here...'
> })})
> ```
> 
> ```css-scss
> .autocomplete-suggestions {
>     background-color: #fff;
>     box-shadow: 0 2px 2px 0 rgba(0,0,0,0.14), 0 1px 5px 0 rgba(0,0,0,0.12), 0 3px 1px -2px rgba(0,0,0,0.2);
>     max-height: 650px;
>     min-width: 100px;
>     overflow-y: auto;
>     will-change: width, height;
>     z-index: 999;
> 
>     .autocomplete-suggestion,
>     .autocomplete-no-suggestion {
>         clear: both;
>         color: #26a69a;
>         color: rgba(0,0,0,0.87);
>         cursor: pointer;
>         display: block;
>         font-size: 16px;
>         line-height: 1.5rem;
>         line-height: 22px;
>         min-height: 50px;
>         padding: 14px 16px;
>         text-align: left;
>         text-transform: none;
>         width: 100%;
> 
>         &.autocomplete-selected {
>             background-color: #eee;
>         }
>     }
> }
> ```

its conflict with materializecss autocomplete.
how to solve that?

**Comment by ghopretz** [comment]
> Btw, it's pretty easy to make [devbridge jquery-autocomplete](https://github.com/devbridge/jQuery-Autocomplete) work with MaterializeCSS
> 
> ```
> <div class="input-field col s12">
>   <input type="hidden" id="autocomplete-value" class="autocomplete">
>   <input type="text" id="autocomplete-input" class="autocomplete">
>   <label for="autocomplete-input">Autocomplete</label>
> </div>
> ```
> 
> ```js
> $('#autocomplete-input').devbridgeAutocomplete({
>     onSelect: function(suggestion) {
>         $('#autocomplete-value').val(suggestion.data.id)
>      },
>      onInvalidateSelection: function() {
>          $('#autocomplete-value').val('')
>       },
>       serviceUrl: '...an url here...'
> })})
> ```
> 
> ```css-scss
> .autocomplete-suggestions {
>     background-color: #fff;
>     box-shadow: 0 2px 2px 0 rgba(0,0,0,0.14), 0 1px 5px 0 rgba(0,0,0,0.12), 0 3px 1px -2px rgba(0,0,0,0.2);
>     max-height: 650px;
>     min-width: 100px;
>     overflow-y: auto;
>     will-change: width, height;
>     z-index: 999;
> 
>     .autocomplete-suggestion,
>     .autocomplete-no-suggestion {
>         clear: both;
>         color: #26a69a;
>         color: rgba(0,0,0,0.87);
>         cursor: pointer;
>         display: block;
>         font-size: 16px;
>         line-height: 1.5rem;
>         line-height: 22px;
>         min-height: 50px;
>         padding: 14px 16px;
>         text-align: left;
>         text-transform: none;
>         width: 100%;
> 
>         &.autocomplete-selected {
>             background-color: #eee;
>         }
>     }
> }
> ```

its conflict with materializecss autocomplete.
how to solve that?
